src/actions/tamanu/prepare_upgrade.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::{bail, IntoDiagnostic, Result, WrapErr};
+use node_semver::Version;
+use regex::Regex;
+use tracing::info;
+
+use crate::actions::{
+	tamanu::{
+		download::{download, make_url},
+		find_package,
+	},
+	Context,
+};
+
+use super::{download::ServerKind, find_existing_version, find_tamanu, TamanuArgs};
+
+/// Setup new Tamanu instllation without taking down the existing one
+#[derive(Debug, Clone, Parser)]
+pub struct PrepareUpgrade {
+	/// Version to update to.
+	#[arg(value_name = "VERSION")]
+	pub version: Version,
+
+	/// Package to upgrade.
+	///
+	/// By default, this command looks for the most recent installed version of Tamanu.
+	/// If both central and facility servers are present and
+	/// configured, it will pick one arbitrarily.
+	#[arg(short, long)]
+	pub package: Option<String>,
+
+	/// Force installing older Tamanu
+	#[arg(long)]
+	pub force: bool,
+}
+
+pub async fn run(ctx: Context<TamanuArgs, PrepareUpgrade>) -> Result<()> {
+	let PrepareUpgrade {
+		version: new_version,
+		package: _,
+		force,
+	} = ctx.args_sub;
+
+	let (_, root) = find_tamanu(&ctx.args_top)?;
+	let existing_version = find_existing_version()?;
+
+	let package = match ctx.args_sub.package {
+		Some(package) => package,
+		None => find_package(&root)?,
+	};
+	info!(?package, "using");
+	let kind = match package.as_str() {
+		"central-server" => ServerKind::Central,
+		"facility-server" => ServerKind::Facility,
+		_ => bail!("package {package} not recognised"),
+	};
+
+	// FIXME: this may not support for Linux as directory structures are different
+	// `root` should have a parent.
+	let existing_root = root
+		.parent()
+		.unwrap()
+		.join(format!("release-v{existing_version}"));
+	let new_root = root
+		.parent()
+		.unwrap()
+		.join(format!("release-v{new_version}"));
+	let new_web_root = root
+		.parent()
+		.unwrap()
+		.join(format!("tamanu-web-{new_version}"));
+
+	let minimum_version = Version::parse("2.0.0").unwrap();
+	if existing_version < minimum_version || new_version < minimum_version {
+		bail!("`PreUpgrade-Tamanu` only support upgrading from/to versions from 2.0.0 onwards");
+	}
+
+	if !force && (new_version <= existing_version) {
+		bail!("the specified version is older than or equal to the installed version");
```

Comment:
```suggestion
		bail!("refusing to downgrade (from {existing_version} to {new_version}) without `--force-downgrade`");
```

Also we should have a separate error for new == existing